src/dscanner/analysis/unused_result.d

```diff
@@ -24,105 +16,212 @@ import std.range : retro;
  * When the return value is intentionally discarded, `cast(void)` can
  * be prepended to silence the check.
  */
-final class UnusedResultChecker : BaseAnalyzer
+extern (C++) class UnusedResultChecker(AST) : BaseAnalyzerDmd
 {
-    alias visit = BaseAnalyzer.visit;
+	alias visit = BaseAnalyzerDmd.visit;
+	mixin AnalyzerInfo!"unused_result";
 
-    mixin AnalyzerInfo!"unused_result";
+	private enum KEY = "dscanner.performance.enum_array_literal";
+	private enum string MSG = "Function return value is discarded";
 
-private:
+	private bool[string] nonVoidFuncs;
+	private string[] aggregateStack;
 
-    enum string KEY = "dscanner.unused_result";
-    enum string MSG = "Function return value is discarded";
+	extern (D) this(string fileName, bool skipTests = false)
+	{
+		super(fileName, skipTests);
+	}
 
-public:
+	private template VisitAggregate(NodeType)
```

Comment:
This is necessary in order to keep the exact same behavior as the previous implementation with libdparse